client/me/help/help-contact/index.jsx

```diff
@@ -418,12 +421,18 @@ const HelpContact = React.createClass( {
 	 */
 	getView: function() {
 		const { olark, confirmation, sitesInitialized, isSubmitting } = this.state;
+		const { isTicketSupportConfigurationReady, isTicketSupportEligible, ticketSupportRequestError } = this.props;
+
 		const showHappychatVariation = this.shouldUseHappychat();
 		const showChatVariation = olark.isUserEligible && olark.isOperatorAvailable;
-		const showKayakoVariation = ! showChatVariation && ( olark.details.isConversing || olark.isUserEligible );
+		const showKayakoVariation = ! showChatVariation && ( olark.details.isConversing || isTicketSupportEligible );
 		const showForumsVariation = ! ( showChatVariation || showKayakoVariation );
 		const showHelpLanguagePrompt = ( olark.locale !== i18n.getLocaleSlug() );
-		const showPreloadForm = ! ( olark.isOlarkReady && sitesInitialized ) && ! this.props.olarkTimedOut;
+
+		const olarkReadyOrTimedOut = olark.isOlarkReady && ! this.props.olarkTimedOut;
+		const ticketReadyOrError = isTicketSupportConfigurationReady || null !== ticketSupportRequestError;
+
+		const showPreloadForm = ! sitesInitialized || ! ticketReadyOrError || ! olarkReadyOrTimedOut ;
```

Comment:
There are quite a lot of tricky boolean operations happening here that could probably be moved into selectors.